app/src/main/java/net/aiscope/gdd_app/dagger/SampleCompletionModule.kt

```diff
@@ -0,0 +1,55 @@
+package net.aiscope.gdd_app.dagger
+
+import android.app.Activity
+import android.content.Context
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import dagger.Binds
+import dagger.Module
+import dagger.Provides
+import net.aiscope.gdd_app.network.RemoteStorage
+import net.aiscope.gdd_app.repository.SampleRepository
+import net.aiscope.gdd_app.ui.sample_completion.SampleCompletionActivity
+import net.aiscope.gdd_app.ui.sample_completion.SampleCompletionViewModel
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.inject.Singleton
+
+@Module
+abstract class SampleCompletionModule {
+
+    @Binds
+    @PerActivity
+    internal abstract fun activity(activity: SampleCompletionActivity): Activity
+
+    @Binds
+    @PerActivity
+    internal abstract fun bindViewModelFactory(factory: SampleCompletionViewModelFactory): ViewModelProvider.Factory
+
+    companion object {
+        @Provides
+        @PerActivity
+        internal fun provideViewModel(
+            repository: SampleRepository,
+            remoteStorage: RemoteStorage,
+            context: Context
+        ): SampleCompletionViewModel =
+            SampleCompletionViewModel(repository, remoteStorage, context)
+    }
+}
+
+@Singleton
+class SampleCompletionViewModelFactory @Inject constructor(
+    private val provider: Provider<SampleCompletionViewModel>
+) : ViewModelProvider.Factory {
+
+    override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
Can create() be overridden in this way ?
```
override fun <T : ViewModel> create(SampleCompletionViewModel: Class<T>): T {
        return provider.get() as T
    }
```

[Reference Doc](https://developer.android.com/reference/androidx/lifecycle/ViewModelProvider.Factory#create(java.lang.Class%3CT%3E))

Reply:
I mean, maybe this is some Kotlin thingy I am unfamiliar with, but that would not prevent you from calling the method with a different class right?

`SampleCompletionModel` simply becomes the new name for 'modelClass' I think (And a pretty confusing one at that)

Reply:
You are right. Embarrassingly, I got confused with an unfamiliar syntax.